### PR TITLE
update set_experiments to filter experiments by channel and app_name

### PR DESCRIPTION
**Pull request by jeddai**
https://mozilla-hub.atlassian.net/browse/EXP-3769

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Review by jhugman — COMMENTED** [review]
A small drive by! Good job!

File: components/nimbus/src/stateless/cirrus_client.rs [hunk 1]
```diff
@@ -163,13 +163,34 @@ impl CirrusClient {
...
+            .iter()
+            .filter_map(|exp| {
+                match exp.app_id.clone().unwrap_or("".into()) == self.app_context.app_id
+                    && exp.channel.clone().unwrap_or("".into()) == self.app_context.channel
+                {
+                    true => Some(exp.clone()),
+                    false => None,
+                }
+            })
+            .collect();
```
[the remark this answers is not in the document]
Comment (jeddai) [reply components/nimbus/src/stateless/cirrus_client.rs:183]: All fantastic points, I'll update it! I didn't realize the `Some` comparison worked as such but in hindsight I probably should've assumed it did or at least tried it and let the unit test double check

File: components/nimbus/src/stateless/cirrus_client.rs [hunk 2]
```diff
@@ -163,13 +163,32 @@ impl CirrusClient {
...
+    /// 2) filters the list of experiments down to only experiments matching the client's `app_name`
+    ///    and `channel`
+    /// 3) writes the resulting value to the client's internal mutable state
     pub fn set_experiments(&self, experiments: String) -> Result<()> {
         let mut state = self.state.lock().unwrap();
-        state.experiments = parse_experiments(&experiments)?;
+        let mut exps: Vec<_> = Default::default();
+        for exp in parse_experiments(&experiments)? {
+            if exp.app_name == Some(self.app_context.app_name.clone())
+                && exp.channel == Some(self.app_context.channel.clone())
```
Comment (jhugman) [inline components/nimbus/src/stateless/cirrus_client.rs:176]: Nit: can we further reduce the number of clones?

```suggestion
            if exp.app_name.as_deref() == Some(&self.app_context.app_name)
                && exp.channel.as_deref() == Some(&self.app_context.channel)
```

File: components/nimbus/src/tests/stateless/test_cirrus_client.rs [hunk 3]
```diff
@@ -143,6 +144,32 @@ fn test_handle_enrollment_errors_on_no_client_id() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_only_writes_experiments_matching_configured_app_id_and_channel() -> Result<()> {
```
Comment (jhugman) [inline components/nimbus/src/tests/stateless/test_cirrus_client.rs:148]: Nit: 
```suggestion
fn test_only_writes_experiments_matching_configured_app_name_and_channel() -> Result<()> {
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
 [Full Changelog](In progress)
 
+## Nimbus SDK ⛅️🔬🔭
+
+### ✨ What's New ✨
+
+- The `set_experiments` method has been updated to filter down the list of experiments to only those that match the configured `app_id` and `channel` ([#5813](https://github.com/mozilla/application-services/pull/5813)).
```
Comment (jhugman) [inline CHANGELOG.md:9]:
```suggestion
- The `set_experiments` method has been updated to filter down the list of experiments to only those that match the configured `app_name` and `channel` ([#5813](https://github.com/mozilla/application-services/pull/5813)). 
```

**Review by jhugman — APPROVED** [review]
Couple of nits, but otherwise, looking good! 🚢

File: components/nimbus/src/stateless/cirrus_client.rs (hunk 2 above)
Comment (jeddai) [reply components/nimbus/src/stateless/cirrus_client.rs]: Ah yeah

**Pull request merged**